### DISCUSSION TITLE
[HELP] How does rendering to solid JS work?

### DISCUSSION THREAD

**Discussion by Gellipapa** [solidjs/solid#1360] [Q&A] [answered]
Hi! First of all, thank you for allowing me to use this wonderful framework.

My question is, for some reason solid js is not rendering my component again and I don't know why. I have tried many different ways but I have not been able to figure out actually how this works in solid js. I have mostly tried to approach it from REACT JS.

Let me give you a simple example of what my problem really is.

```js
//App component
     const [micAndRadio,setMicAndRadio] = createSignal()
     const value = {mic: false, radio: true}
     setMicAndRadio(value)
     <InfoPanel micAndRadio={micAndRadio} />

//InfoPanel component
export const InfoPanel = (props) => {
    return (
       <RadioIcon active={props.micAndRadio()?.radio}/>
   )
}

//RadioIcon component
const RadioIcon = (props) =>{
    console.log(props) // first render empty and then it is not updated whole component
    let currentColor = props.active ? '#F4C900' : 'white'
    return(
       <svg fill={currentColor}></svg>
   )
}
```
The signal value is continuously set but the RadioIcon component does not render again.

I've tried createStore too. I don't really understand how the renderer actually works then because I don't use destruct them I just pass them as props anyway. 

Can anyone help me? Thank you.

**Comment by maxmilton** [comment]
Have a read of <https://www.solidjs.com/guides/reactivity#considerations> and pay particular attention to `2.`.

**Reply by Gellipapa** [reply]
Thank you very much for your reply. 

If I understand correctly, I can't send it as a prop anymore, but I have to use createEffect or another createSignal to handle the reactivity. 

I tried this in part of the code to createEffect and solve it that way, but for me it was so strange that I can't easily pass the prop value, but I have to create another variable/signal.

Maybe you could show me an example of how I should do it then to understand it better?

**Comment by atk** [comment, marked as answer]
No, for derived state, use functions instead of let.

```ts
const currentColor = () => props.active ? '#F4C900' : 'white'
return(
       <svg fill={currentColor()}></svg>
   )
```

**Reply by Gellipapa** [reply]
Hi! Thank you very much for your reply and the example. Should I use it in all these cases and forget the let version completely?
